controlplane/kubeadm/controllers/scale.go

```diff
@@ -64,8 +64,8 @@ func (r *KubeadmControlPlaneReconciler) scaleUpControlPlane(ctx context.Context,
 	logger := controlPlane.Logger()
 
 	// reconcileHealth returns err if there is a machine being delete which is a required condition to check before scaling up
-	if err := r.reconcileHealth(ctx, cluster, kcp, controlPlane); err != nil {
-		return ctrl.Result{}, &capierrors.RequeueAfterError{RequeueAfter: healthCheckFailedRequeueAfter}
+	if result, err := r.reconcileHealth(ctx, cluster, kcp, controlPlane); !result.IsZero() || err != nil {
+		return result, err
 	}
```

Comment:
Alternatively, I'm wondering if it might make things easier to read and more explicit if we check for each condition independently:
```suggestion
	result, err := r.reconcileHealth(ctx, cluster, kcp, controlPlane)
	if err != nil {
		return ctrl.Result{}, err
	} else if !result.IsZero() {
		return result, nil
	}
```

Reply:
I'm fine either way, @ncdc has a special place for else conditions though :P

Reply:
Right, the `else` here is not required 😄